node/src/accountant/mod.rs

```diff
@@ -652,26 +704,48 @@ impl Accountant {
     }
 
     fn handle_payable_payment_setup(&mut self, msg: BlockchainAgentWithContextMessage) {
-        let blockchain_bridge_instructions = match self
+        let response_skeleton_opt = msg.response_skeleton_opt;
+        let blockchain_bridge_instructions_opt = match self
             .scanners
             .payable
             .try_skipping_payment_adjustment(msg, &self.logger)
         {
-            Ok(Either::Left(finalized_msg)) => finalized_msg,
-            Ok(Either::Right(unaccepted_msg)) => {
+            Some(Either::Left(complete_msg)) => Some(complete_msg),
+            Some(Either::Right(unaccepted_msg)) => {
```

Comment:
It's a bit concerning to me that this combination of match statement and the possibility of the `perform_payment_adjustment` can return an `Option<T>`.

This opens up a lot of cases, and we would have to maintain all those cases for the lifetime of this code.

My assumption was that we need to take care of the adjustment in a single step, and that way it'll just return a `Result<>`. On the other hand there was the concern that if we do that we'll have to manage the Actor messages between Neighbourhood with the PaymentAdjuster. This concern balances out and keeps us in dilemma what should be the right step.

Since, I had a lack of info at the time of writing this message, please forgive me if this comment seems undesirable, and I'd appreciate if by any chance the comment seems reasonable, please investigate.